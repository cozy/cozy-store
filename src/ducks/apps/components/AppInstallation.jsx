import React, { Component } from 'react'

import { connect } from 'react-redux'
import { PropTypes } from 'react-proptypes'
import flags from 'cozy-flags'

import ReactMarkdownWrapper from 'ducks/components/ReactMarkdownWrapper'
import { ModalDescription, ModalHeader, ModalFooter } from 'cozy-ui/react/Modal'
import Spinner from 'cozy-ui/react/Spinner'
import Button from 'cozy-ui/react/Button'

import PermissionsList from 'ducks/apps/components/PermissionsList'
import Partnership from 'ducks/apps/components/Partnership'
import { translate } from 'cozy-ui/react/I18n'
import Alerter from 'cozy-ui/react/Alerter'
import Checkbox from 'cozy-ui/react/Checkbox'

import { getTranslatedManifestProperty } from 'lib/helpers'
import { hasPendingUpdate } from 'ducks/apps/appStatus'
import storeConfig from 'config'

import { APP_TYPE, getAppBySlug, installAppFromRegistry } from 'ducks/apps'

const shouldSkipPermissions = app =>
  flags('skip-low-permissions') &&
  app.label <= storeConfig.default.authorizedLabelLimit

const hasInstallation = app => !app.installed || hasPendingUpdate(app)

export class AppInstallation extends Component {
  constructor(props) {
    super(props)
    const { app } = props
    this.state = { isTermsAccepted: false }
    this.installApp = this.installApp.bind(this)
    if (shouldSkipPermissions(app) && hasInstallation(app)) {
      this.installApp()
    }
  }

  installApp() {
    if (!this.isInstallReady()) return // Not ready to be installed
    const { app, channel, installApp, onInstallOrUpdate, t } = this.props
    const isUpdate = app.installed
    const isChannelSwitch = !!channel
    installApp(app, channel, isUpdate).catch(error => {
      const appName = getTranslatedManifestProperty(app, 'name', t)
      Alerter.error(
        t('app_modal.install.alert.install_error', {
          message: error.message,
          name: appName
        }),
        {
          duration: 12000,
          buttonText: t('app_modal.install.alert.dismiss'),
          buttonAction: dismiss => dismiss()
        }
      )
      console.error(error)
    })
    if (isUpdate || isChannelSwitch || app.type !== APP_TYPE.KONNECTOR) {
      if (typeof onInstallOrUpdate === 'function') onInstallOrUpdate()
    }
  }

  componentDidUpdate = prevProps => {
    const { app, onKonnectorInstall } = this.props
    const justInstalled =
      !!prevProps.app && !prevProps.app.installed && app.installed
    if (justInstalled && app.type === APP_TYPE.KONNECTOR) {
      if (typeof onKonnectorInstall === 'function') onKonnectorInstall()
    }
  }

  acceptTerms = () => {
    this.setState(state => ({
      isTermsAccepted: !state.isTermsAccepted
    }))
  }

  isInstallReady = () => {
    const { app, isInstalling } = this.props
    const isCurrentAppInstalling = isInstalling === app.slug
    const isTermsReady =
      shouldSkipPermissions(app) || (!app.terms || this.state.isTermsAccepted)
    return !isCurrentAppInstalling && isTermsReady
  }

  render() {
    const {
      app,
      fetchError,
      installError,
      isFetching,
      isAppFetching,
      isInstalling,
      onCancel,
      t
    } = this.props
    const { isTermsAccepted } = this.state
    const isFetchingSomething = isFetching || isAppFetching
    const isCurrentAppInstalling = isInstalling === app.slug

    if (shouldSkipPermissions(app) && hasInstallation(app)) {
      return (
        <React.Fragment>
          <ModalDescription className="sto-install-loading-wrapper">
            <div className="sto-install-loading">
              <Spinner size="xxlarge" />
            </div>
          </ModalDescription>
        </React.Fragment>
      )
    }

    return (
      <React.Fragment>
        <ModalHeader title={t('app_modal.install.title')} />
        {isFetchingSomething ? (
          <ModalDescription className="sto-install-loading-wrapper">
            <div className="sto-install-loading">
              <Spinner size="xlarge" />
            </div>
          </ModalDescription>
        ) : (
          <ModalDescription>
            {app.partnership && <Partnership app={app} />}
            {app.permissions && <PermissionsList app={app} />}
            {fetchError && (
              <p className="u-error">
                {t('app_modal.install.message.version_error', {
                  message: fetchError.message
                })}
              </p>
            )}
          </ModalDescription>
        )}
        {!isFetchingSomething && !fetchError && (
          <ModalFooter className="sto-install-footer">
            {installError && (
              <p className="u-error">
                {t('app_modal.install.message.install_error', {
                  message: installError.message
                })}
              </p>
            )}
            {app.terms && (
              <div className="sto-install-terms">
                <Checkbox
                  className="sto-install-terms-checkbox"
                  onChange={this.acceptTerms}
                  checked={isTermsAccepted}
                  disabled={isInstalling}
                >
                  <ReactMarkdownWrapper
                    source={t('app_modal.install.terms', {
                      url: app.terms.url
                    })}
                  />
                </Checkbox>
              </div>
            )}
            <div className="sto-install-controls">
              <Button
                theme="secondary"
                onClick={onCancel}
                disabled={isCurrentAppInstalling}
                label={t('app_modal.install.cancel')}
                extension="full"
                className="u-mh-half"
              />
              <Button
                theme="primary"
                disabled={!this.isInstallReady() || isInstalling}
                busy={isCurrentAppInstalling}
                extension="full"
                onClick={this.installApp}
                label={
                  hasPendingUpdate(app)
                    ? t('app_modal.install.update')
                    : t('app_modal.install.install')
                }
                className="u-mh-half"
              />
            </div>
          </ModalFooter>
        )}
      </React.Fragment>
    )
  }
}

AppInstallation.propTypes = {
  appSlug: PropTypes.string.isRequired
}

const mapStateToProps = (state, ownProps) => ({
  app: getAppBySlug(state, ownProps.appSlug),
  isFetching: state.apps.isFetching,
  isAppFetching: state.apps.isAppFetching,
  isInstalling: state.apps.isInstalling,
  fetchError: state.apps.fetchError,
  installError: state.apps.actionError
})

const mapDispatchToProps = dispatch => ({
  installApp: (app, channel, isUpdate) =>
    dispatch(installAppFromRegistry(app, channel, isUpdate))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(translate()(AppInstallation))
