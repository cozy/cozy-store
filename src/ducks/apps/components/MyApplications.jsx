/* global cozy */
import React, { Component } from 'react'
import { useMatch } from 'react-router-dom'

import { translate } from 'cozy-ui/transpiled/react/I18n'
import withBreakpoints from 'cozy-ui/transpiled/react/helpers/withBreakpoints'
import { Content } from 'cozy-ui/transpiled/react/Layout'

import ApplicationRouting from 'ducks/apps/components/ApplicationRouting'
import Sections from 'ducks/apps/components/QuerystringSections'
import AppsLoading from 'ducks/components/AppsLoading'

import { useNavigateNoUpdates, withRouterUtils } from 'lib/RouterUtils'

const { BarCenter } = cozy.bar

export class MyApplications extends Component {
  constructor(props) {
    super(props)
    this.onAppClick = this.onAppClick.bind(this)
  }

  onAppClick(appSlug) {
    this.props.navigate(`/myapps/${appSlug}`)
  }

  render() {
    const {
      t,
      installedApps,
      isFetching,
      isAppFetching,
      fetchError,
      actionError,
      breakpoints = {},
      isExact
    } = this.props
    const { isMobile } = breakpoints
    const title = <h2 className="sto-view-title">{t('myapps.title')}</h2>
    return (
      <Content className="sto-myapps">
        {isExact && isFetching && <AppsLoading />}
        <div className="sto-list-container">
          {isMobile && <BarCenter>{title}</BarCenter>}
          {!isFetching && (
            <Sections
              apps={installedApps}
              error={fetchError}
              onAppClick={this.onAppClick}
            />
          )}
        </div>

        <ApplicationRouting
          installedApps={installedApps}
          isFetching={isFetching}
          isAppFetching={isAppFetching}
          actionError={actionError}
          parent="myapps"
        />
      </Content>
    )
  }
}

const MyApplicationsWrapper = props => {
  const isExact = useMatch('myapps')
  const navigate = useNavigateNoUpdates()

  return <MyApplications {...props} isExact={isExact} navigate={navigate} />
}

export default translate()(
  withBreakpoints()(withRouterUtils(MyApplicationsWrapper))
)
