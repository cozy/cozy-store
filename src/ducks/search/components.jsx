import StoreAppItem from 'ducks/apps/components/StoreAppItem'
import { dumpMatches } from 'ducks/search/utils'
import sortBy from 'lodash/sortBy'
import PropTypes from 'prop-types'
import React, { useCallback, useMemo } from 'react'

import flag from 'cozy-flags'
import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import Icon from 'cozy-ui/transpiled/react/Icon'
import Magnifier from 'cozy-ui/transpiled/react/Icons/Magnifier'
import Input from 'cozy-ui/transpiled/react/Input'
import InputGroup from 'cozy-ui/transpiled/react/InputGroup'
import Label from 'cozy-ui/transpiled/react/Label'
import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'

export const SearchField = ({ onChange, value }) => {
  const { t } = useI18n()
  const { isMobile } = useBreakpoints()
  const handleChange = useCallback(
    ev => {
      onChange(ev.target.value)
    },
    [onChange]
  )
  return (
    <div className="u-flex u-flex-items-center u-flex-grow-1">
      {!isMobile ? (
        <Label className="u-di u-mr-half" htmlFor="discover-search">
          {t('discover-search-field.label')}
        </Label>
      ) : null}
      <InputGroup
        prepend={
          isMobile ? (
            <Icon icon={Magnifier} className="u-pl-1 u-coolGrey" />
          ) : null
        }
      >
        <Input
          id="discover-search"
          placeholder={t('discover-search-field.placeholder')}
          onChange={handleChange}
          type="text"
          value={value}
        />
      </InputGroup>
    </div>
  )
}

export const SearchResults = ({ searchResults, onAppClick, disableClick }) => {
  const sortedSortResults = useMemo(() => {
    return sortBy(searchResults, result => result.score)
  }, [searchResults])
  return (
    <div className="u-mv-1 u-flex u-flex-wrap">
      {sortedSortResults.map(result => {
        const app = result.item
        const isDisableClick = disableClick?.(app)
        return flag('store.show-search-score') ? (
          <div>
            <StoreAppItem
              onClick={() => !isDisableClick && onAppClick(app.slug)}
              key={app.slug}
              app={app}
            />
            <small
              onClick={() => dumpMatches(result)}
              title={JSON.stringify(result.matches, null, 2)}
            >
              {result.score.toFixed(2)}
            </small>
          </div>
        ) : (
          <StoreAppItem
            onClick={() => !isDisableClick && onAppClick(app.slug)}
            key={app.slug}
            app={app}
          />
        )
      })}
    </div>
  )
}

SearchResults.propTypes = {
  onAppClick: PropTypes.func.isRequired,
  disableClick: PropTypes.func,
  searchResults: PropTypes.array
}
