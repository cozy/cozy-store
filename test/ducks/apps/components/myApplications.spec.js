'use strict'

import { MyApplications } from 'ducks/apps/components/MyApplications'
import { shallow } from 'enzyme'
import React from 'react'

import { BarCenter } from 'cozy-bar'

import { tMock } from '../../../jestLib/I18n'
import mockApps from '../_mockApps'

const mockMyApplicationsError = new Error('This is a test error')

const mockInstalledApps = mockApps.filter(a => a.installed)

const getMockProps = (
  installedApps = mockInstalledApps,
  isFetching = false,
  fetchError = null
) => ({
  fetchApps: jest.fn(),
  uninstallApp: jest.fn(),
  installedApps,
  isFetching,
  fetchError,
  actionError: null,
  navigate: jest.fn(),
  searchParams: new URLSearchParams()
})

describe('MyApplications component', () => {
  it('should be rendered correctly if apps uninstalled', () => {
    const mockProps = getMockProps()
    const component = shallow(
      <MyApplications t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly with apps and URL search params', () => {
    // default values so use null
    const mockProps = getMockProps(
      mockInstalledApps,
      false,
      null,
      { isExact: true },
      { search: '?category=cozy' }
    )
    const component = shallow(
      <MyApplications t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should render correctly a spinner if apps is fetching', () => {
    const mockProps = getMockProps([], true, null)
    const component = shallow(
      <MyApplications t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should display error from props correctly', () => {
    const mockProps = getMockProps([], false, mockMyApplicationsError)
    const component = shallow(
      <MyApplications t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should define the correct onAppClick function to pass to sections', () => {
    const mockProps = getMockProps()
    const component = shallow(<MyApplications t={tMock} {...mockProps} />)
    const instance = component.instance()
    instance.onAppClick(mockInstalledApps[0].slug)
    expect(mockProps.navigate).toHaveBeenCalledTimes(1)
    expect(mockProps.navigate).toHaveBeenCalledWith(
      `/myapps/${mockInstalledApps[0].slug}`
    )
  })

  it('should use BarCenter from cozy-bar in mobile view only', () => {
    const mockProps = getMockProps()
    const component = shallow(<MyApplications t={tMock} {...mockProps} />)
    expect(component.find(BarCenter).length).toBe(0)
    const componentMobile = shallow(
      <MyApplications
        t={tMock}
        breakpoints={{ isMobile: true }}
        {...mockProps}
      />
    )
    expect(componentMobile.find(BarCenter).length).toBe(1)
    expect(componentMobile.getElement()).toMatchSnapshot()
  })

  it('should transfer search params when navigating to app page', () => {
    const mockProps = getMockProps()
    mockProps.searchParams.set('type', 'konnector')
    mockProps.searchParams.set('category', 'isp')
    mockProps.searchParams.size = 2 // Bug: URLSearchParams have undefined size when run in tests so we have to enforce it
    const component = shallow(<MyApplications t={tMock} {...mockProps} />)
    const instance = component.instance()
    instance.onAppClick(mockInstalledApps[0].slug)
    expect(mockProps.navigate).toHaveBeenCalledTimes(1)
    expect(mockProps.navigate).toHaveBeenCalledWith(
      `/myapps/collect?type=konnector&category=isp`
    )
  })
})
