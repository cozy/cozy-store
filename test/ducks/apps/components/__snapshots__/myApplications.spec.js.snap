// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyApplications component should be rendered correctly if apps uninstalled 1`] = `
<Content
  className="sto-myapps"
>
  <div
    className="sto-list-container"
  >
    <QuerystringSections
      apps={
        Array [
          Object {
            "categories": Array [
              "cozy",
            ],
            "developer": Object {
              "name": "Cozy",
            },
            "editor": "Cozy",
            "icon": "<svg></svg>",
            "installed": true,
            "isInRegistry": true,
            "label": 1,
            "name": "Collect",
            "name_prefix": "Cozy",
            "permissions": Object {
              "mock": Object {
                "type": "io.mock.doctype",
              },
              "mock2": Object {
                "type": "io.mock.doctype2",
              },
            },
            "related": "http://collect.cozy.mock/",
            "slug": "collect",
            "tags": Array [
              "konnector",
              "collect",
              "bills",
              "providers",
              "files",
            ],
            "type": "webapp",
            "uninstallable": false,
            "version": "3.0.0",
            "versions": Object {
              "beta": Array [
                "3.0.0",
              ],
              "dev": Array [
                "3.0.0",
              ],
              "stable": Array [
                "3.0.0",
              ],
            },
          },
          Object {
            "categories": Array [
              "cozy",
            ],
            "developer": Object {
              "name": "Cozy",
            },
            "editor": "Cozy",
            "icon": "<svg></svg>",
            "installed": true,
            "name": "Drive",
            "name_prefix": "Cozy",
            "permissions": Object {
              "mock2": Object {
                "type": "io.mock.doctype2",
              },
            },
            "related": "http://drive.cozy.mock/",
            "slug": "drive",
            "tags": Array [
              "search",
              "files",
              "folders",
            ],
            "type": "webapp",
            "uninstallable": false,
            "version": "3.0.0-beta89bnhj3993",
          },
          Object {
            "categories": Array [
              "transport",
            ],
            "developer": Object {
              "name": "Cozy",
            },
            "icon": "<svg></svg>",
            "installed": true,
            "isInRegistry": true,
            "name": "Trinlane",
            "permissions": Object {
              "mock": Object {
                "type": "io.mock.doctype",
              },
            },
            "slug": "konnector-trinlane",
            "tags": Array [
              "transport",
              "files",
              "bills",
            ],
            "type": "konnector",
            "uninstallable": true,
            "version": "0.1.0",
            "versions": Object {
              "beta": Array [
                "0.1.0",
              ],
              "dev": Array [
                "0.1.0",
              ],
              "stable": Array [
                "0.1.0",
              ],
            },
          },
          Object {
            "availableVersion": "3.0.0",
            "categories": Array [
              "partners",
            ],
            "icon": "<svg></svg>",
            "installed": true,
            "isInRegistry": true,
            "name": "Tasky",
            "slug": "tasky",
            "type": "webapp",
            "uninstallable": true,
            "versions": Object {
              "beta": Array [
                "3.0.0",
                "1.0.0",
              ],
              "dev": Array [
                "3.0.0",
                "1.0.0",
              ],
              "stable": Array [
                "3.0.0",
                "1.0.0",
              ],
            },
          },
        ]
      }
      error={null}
      onAppClick={[Function]}
    />
  </div>
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    installedApps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "label": 1,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "name": "Drive",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://drive.cozy.mock/",
          "slug": "drive",
          "tags": Array [
            "search",
            "files",
            "folders",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0-beta89bnhj3993",
        },
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
        Object {
          "availableVersion": "3.0.0",
          "categories": Array [
            "partners",
          ],
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Tasky",
          "slug": "tasky",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
              "1.0.0",
            ],
            "dev": Array [
              "3.0.0",
              "1.0.0",
            ],
            "stable": Array [
              "3.0.0",
              "1.0.0",
            ],
          },
        },
      ]
    }
    isFetching={false}
    parent="myapps"
  />
</Content>
`;

exports[`MyApplications component should be rendered correctly with apps and URL search params 1`] = `
<Content
  className="sto-myapps"
>
  <div
    className="sto-list-container"
  >
    <QuerystringSections
      apps={
        Array [
          Object {
            "categories": Array [
              "cozy",
            ],
            "developer": Object {
              "name": "Cozy",
            },
            "editor": "Cozy",
            "icon": "<svg></svg>",
            "installed": true,
            "isInRegistry": true,
            "label": 1,
            "name": "Collect",
            "name_prefix": "Cozy",
            "permissions": Object {
              "mock": Object {
                "type": "io.mock.doctype",
              },
              "mock2": Object {
                "type": "io.mock.doctype2",
              },
            },
            "related": "http://collect.cozy.mock/",
            "slug": "collect",
            "tags": Array [
              "konnector",
              "collect",
              "bills",
              "providers",
              "files",
            ],
            "type": "webapp",
            "uninstallable": false,
            "version": "3.0.0",
            "versions": Object {
              "beta": Array [
                "3.0.0",
              ],
              "dev": Array [
                "3.0.0",
              ],
              "stable": Array [
                "3.0.0",
              ],
            },
          },
          Object {
            "categories": Array [
              "cozy",
            ],
            "developer": Object {
              "name": "Cozy",
            },
            "editor": "Cozy",
            "icon": "<svg></svg>",
            "installed": true,
            "name": "Drive",
            "name_prefix": "Cozy",
            "permissions": Object {
              "mock2": Object {
                "type": "io.mock.doctype2",
              },
            },
            "related": "http://drive.cozy.mock/",
            "slug": "drive",
            "tags": Array [
              "search",
              "files",
              "folders",
            ],
            "type": "webapp",
            "uninstallable": false,
            "version": "3.0.0-beta89bnhj3993",
          },
          Object {
            "categories": Array [
              "transport",
            ],
            "developer": Object {
              "name": "Cozy",
            },
            "icon": "<svg></svg>",
            "installed": true,
            "isInRegistry": true,
            "name": "Trinlane",
            "permissions": Object {
              "mock": Object {
                "type": "io.mock.doctype",
              },
            },
            "slug": "konnector-trinlane",
            "tags": Array [
              "transport",
              "files",
              "bills",
            ],
            "type": "konnector",
            "uninstallable": true,
            "version": "0.1.0",
            "versions": Object {
              "beta": Array [
                "0.1.0",
              ],
              "dev": Array [
                "0.1.0",
              ],
              "stable": Array [
                "0.1.0",
              ],
            },
          },
          Object {
            "availableVersion": "3.0.0",
            "categories": Array [
              "partners",
            ],
            "icon": "<svg></svg>",
            "installed": true,
            "isInRegistry": true,
            "name": "Tasky",
            "slug": "tasky",
            "type": "webapp",
            "uninstallable": true,
            "versions": Object {
              "beta": Array [
                "3.0.0",
                "1.0.0",
              ],
              "dev": Array [
                "3.0.0",
                "1.0.0",
              ],
              "stable": Array [
                "3.0.0",
                "1.0.0",
              ],
            },
          },
        ]
      }
      error={null}
      onAppClick={[Function]}
    />
  </div>
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    installedApps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "label": 1,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "name": "Drive",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://drive.cozy.mock/",
          "slug": "drive",
          "tags": Array [
            "search",
            "files",
            "folders",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0-beta89bnhj3993",
        },
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
        Object {
          "availableVersion": "3.0.0",
          "categories": Array [
            "partners",
          ],
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Tasky",
          "slug": "tasky",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
              "1.0.0",
            ],
            "dev": Array [
              "3.0.0",
              "1.0.0",
            ],
            "stable": Array [
              "3.0.0",
              "1.0.0",
            ],
          },
        },
      ]
    }
    isFetching={false}
    parent="myapps"
  />
</Content>
`;

exports[`MyApplications component should display error from props correctly 1`] = `
<Content
  className="sto-myapps"
>
  <div
    className="sto-list-container"
  >
    <QuerystringSections
      apps={Array []}
      error={[Error: This is a test error]}
      onAppClick={[Function]}
    />
  </div>
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    installedApps={Array []}
    isFetching={false}
    parent="myapps"
  />
</Content>
`;

exports[`MyApplications component should render correctly a spinner if apps is fetching 1`] = `
<Content
  className="sto-myapps"
>
  <div
    className="sto-list-container"
  />
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    installedApps={Array []}
    isFetching={true}
    parent="myapps"
  />
</Content>
`;

exports[`MyApplications component should use BarCenter from cozy-bar in mobile view only 1`] = `
<Content
  className="sto-myapps"
>
  <div
    className="sto-list-container"
  >
    <BarCenter>
      <h2
        className="sto-view-title"
      >
        Installed
      </h2>
    </BarCenter>
    <QuerystringSections
      apps={
        Array [
          Object {
            "categories": Array [
              "cozy",
            ],
            "developer": Object {
              "name": "Cozy",
            },
            "editor": "Cozy",
            "icon": "<svg></svg>",
            "installed": true,
            "isInRegistry": true,
            "label": 1,
            "name": "Collect",
            "name_prefix": "Cozy",
            "permissions": Object {
              "mock": Object {
                "type": "io.mock.doctype",
              },
              "mock2": Object {
                "type": "io.mock.doctype2",
              },
            },
            "related": "http://collect.cozy.mock/",
            "slug": "collect",
            "tags": Array [
              "konnector",
              "collect",
              "bills",
              "providers",
              "files",
            ],
            "type": "webapp",
            "uninstallable": false,
            "version": "3.0.0",
            "versions": Object {
              "beta": Array [
                "3.0.0",
              ],
              "dev": Array [
                "3.0.0",
              ],
              "stable": Array [
                "3.0.0",
              ],
            },
          },
          Object {
            "categories": Array [
              "cozy",
            ],
            "developer": Object {
              "name": "Cozy",
            },
            "editor": "Cozy",
            "icon": "<svg></svg>",
            "installed": true,
            "name": "Drive",
            "name_prefix": "Cozy",
            "permissions": Object {
              "mock2": Object {
                "type": "io.mock.doctype2",
              },
            },
            "related": "http://drive.cozy.mock/",
            "slug": "drive",
            "tags": Array [
              "search",
              "files",
              "folders",
            ],
            "type": "webapp",
            "uninstallable": false,
            "version": "3.0.0-beta89bnhj3993",
          },
          Object {
            "categories": Array [
              "transport",
            ],
            "developer": Object {
              "name": "Cozy",
            },
            "icon": "<svg></svg>",
            "installed": true,
            "isInRegistry": true,
            "name": "Trinlane",
            "permissions": Object {
              "mock": Object {
                "type": "io.mock.doctype",
              },
            },
            "slug": "konnector-trinlane",
            "tags": Array [
              "transport",
              "files",
              "bills",
            ],
            "type": "konnector",
            "uninstallable": true,
            "version": "0.1.0",
            "versions": Object {
              "beta": Array [
                "0.1.0",
              ],
              "dev": Array [
                "0.1.0",
              ],
              "stable": Array [
                "0.1.0",
              ],
            },
          },
          Object {
            "availableVersion": "3.0.0",
            "categories": Array [
              "partners",
            ],
            "icon": "<svg></svg>",
            "installed": true,
            "isInRegistry": true,
            "name": "Tasky",
            "slug": "tasky",
            "type": "webapp",
            "uninstallable": true,
            "versions": Object {
              "beta": Array [
                "3.0.0",
                "1.0.0",
              ],
              "dev": Array [
                "3.0.0",
                "1.0.0",
              ],
              "stable": Array [
                "3.0.0",
                "1.0.0",
              ],
            },
          },
        ]
      }
      error={null}
      onAppClick={[Function]}
    />
  </div>
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    installedApps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "label": 1,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "name": "Drive",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://drive.cozy.mock/",
          "slug": "drive",
          "tags": Array [
            "search",
            "files",
            "folders",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0-beta89bnhj3993",
        },
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
        Object {
          "availableVersion": "3.0.0",
          "categories": Array [
            "partners",
          ],
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Tasky",
          "slug": "tasky",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
              "1.0.0",
            ],
            "dev": Array [
              "3.0.0",
              "1.0.0",
            ],
            "stable": Array [
              "3.0.0",
              "1.0.0",
            ],
          },
        },
      ]
    }
    isFetching={false}
    parent="myapps"
  />
</Content>
`;
